build/tools/Automation/DotNet/DotNetAutomator.cs

```diff
@@ -58,8 +58,22 @@ public async Task RunAsync()
                 this.oryxSdkStorageBaseUrl = Constants.OryxSdkStorageBaseUrl;
             }
 
-            this.oryxDotNetSdkVersions = await this.httpService.GetOryxSdkVersionsAsync(
-                this.oryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl);
+            string sdkVersionsUrl = this.oryxSdkStorageBaseUrl + DotNetConstants.DotNetSuffixUrl;
+
+            // A SAS token is required for the staging account.
+            if (this.oryxSdkStorageBaseUrl == Constants.OryxSdkStagingStorageBaseUrl)
+            {
+                string sasToken = Environment.GetEnvironmentVariable(Constants.OryxSdkStagingPrivateSasTokenEnvVar);
+                if (string.IsNullOrEmpty(sasToken))
+                {
+                    throw new ArgumentException($"The environment variable {Constants.OryxSdkStagingPrivateSasTokenEnvVar} " +
+                        $"must be provided in order to access {Constants.OryxSdkStagingStorageBaseUrl}");
+                }
+
+                sdkVersionsUrl += "&" + sasToken;
```

Comment:
is it needed to add the parameter separator `&` in between the url and token?

Reply:
Yes, because the URL contains other multiple parameters, for accessing the list of SDKs. Example: https://oryxsdksprod.blob.core.windows.net/python?`restype=container&comp=list&include=metadata`